### ISSUE TITLE
Configuration workflow is not respecting deletes (bcim/bcex).

### ISSUE THREAD

**Issue by serundeputy** [backdrop-contrib/drush#109]
When working on a site make configuration changes that involve removal of config files, like delete a layout.

* Delete layout
* export: `drush bcex`
* commit and push changes
* pull down changes (on server or another dev in the team) issue: `drush bcim`

This works fine in that the dev team will receive the config updates, but when the 2nd dev exports, the deleted layout was not detected (i suspect by the `bcim` step) and on the next config change they export will re-introduce the 'deleted' config (layout in this example) back into the codebase and config :(

* look at `bcim` command and get it to respect deleted files.

----
PR https://github.com/backdrop-contrib/drush/pull/211 by @BWPanda

**Comment by jenlampton** [comment]
It think this might have more to do with how they are using Git. It sounds like they forgot to `git rm` the files that were removed by the sync. It happens to me sometimes too ;)

**Comment by ghost** [comment]
I'm having an issue where uninstalled module's config files are still showing up when I export config using drush.

I think the problem (at least for me) is that drush isn't emptying the 'staging' folder first, so the old config files are still there after the exported ones are added in.

I know there's code in drush to empty the staging folder on export, but it looks like there's a bug and it's not working:

```php
$destination = drush_get_option('destination', '');
// Determine where to export the config.
if (empty($destination)) {
  $config_export = BACKDROP_ROOT . "/" . $config_directories['staging'];
}
else {
  $config_export = BACKDROP_ROOT . "/$destination";
  drush_print_r("\n\n\t\tmade it to destination: $config_export\n\n");
}

passthru("rm -rf " . BACKDROP_ROOT . "/$config_export/");
```
(Source: https://github.com/backdrop-contrib/drush/blob/1.x-1.x/commands/config/backdrop_config_export.drush.inc)

`$config_export` is set to a value that includes `BACKDROP_ROOT`, but when removing files from staging, it includes the `BACKDROP_ROOT` path again. So it's trying to remove: `BACKDROP_ROOT/BACKDROP_ROOT/[PATH_TO_STAGING]/`

**Comment by ghost** [comment]
Hacking drush to use this code instead works for me:

```php
$destination = drush_get_option('destination', '');
// Determine where to export the config.
if (empty($destination)) {
  $config_export = $config_directories['staging'];
}
else {
  $config_export = $destination;
  drush_print_r("\n\n\t\tmade it to destination: $config_export\n\n");
}

passthru("rm -rf " . BACKDROP_ROOT . "/$config_export/");
```

**Comment by ghost** [comment]
Ah heck, here's a PR (made via the GitHub UI): https://github.com/backdrop-contrib/drush/pull/211

**Comment by serundeputy** [comment]
this is great! thanks so much; I'll need to test it; and I should see what if any tests I have going on for `bcim` and `bcex`, but I won't let that hold up this bug fix.

**Comment by serundeputy** [comment]
merged will ship with `1.2.0`. thanks @BWPanda !
